### ISSUE TITLE
redis doesnt work on v20170321.0

### ISSUE THREAD

**Issue by grauschnabel** [flynn/flynn#4014]
Hi there,

I have odoo instances on flynn app, with flynn-odoo repo on github (myown).
On flynn v20170217.0 it worked fine, but on v20170321.0 it tells me in the logs:
`[2017-03-26T13:10:07.003708Z]    ConnectionError: Error -2 connecting to leader.redis-1a87d1e4-9c10-4852-a2dd-680fa75e4247:6379. Name or service not known.
`
when i remove the resource odoo works fine, but it should not run save by more than one processes.  So please tell me what to do...
thanks,
Martin

**Comment by lmars** [comment]
@grauschnabel did you restore the `v20170321.0` cluster from a backup or is it a fresh cluster?

Does the `redis-1a87d1e4-9c10-4852-a2dd-680fa75e4247` app exist (check `flynn apps`).

**Comment by grauschnabel** [comment]
1. It's a fresh cluster.
2. I will look at that tomorrow, but "flynn redis redis-cli" worked, I remember.

**Comment by grauschnabel** [comment]
Yes, the redis-*** app exists.  How can I fix this?  This is a breaking bug for me.

**Comment by lmars** [comment]
@grauschnabel the Redis appliance is currently broken as it generates incorrect `REDIS_HOST` and `REDIS_URL` environment variables.

I have proposed a fix in #4015 but in the meantime you can manually fix them by setting `$APP` to the name of your app and running:

```
$ source <(flynn -a $APP env)
$ flynn -a $APP env set \
  REDIS_HOST="leader.${FLYNN_REDIS}.discoverd" \
  REDIS_URL="redis://:${REDIS_PASSWORD}@leader.${FLYNN_REDIS}.discoverd:6379"
```

**Comment by grauschnabel** [comment]
thanks works fine
